TR review feedback for transform chapters (#1310)
hadley/r4ds: numbers.qmd

```

> And you've computed the time between each events, and figured out if there's a gap that's big enough to qualify:

Reviewer: ```suggestion
And you've computed the time between each event, and figured out if there's a gap that's big enough to qualify:
```

```{r}
events <- events |> 
  mutate(
    diff = time - lag(time, default = first(time)),
    has_gap = diff >= 5
  )
events
```

But how do we go from that logical vector to something that we can `group_by()`?

> `cumsum()`, from @sec-cumulative-and-rolling-aggregates, comes to the rescue as gap, i.e. `has_gap` is `TRUE`, will increments `group` by one (@sec-numeric-summaries-of-logicals):

Reviewer: ```suggestion
`cumsum()`, from @sec-cumulative-and-rolling-aggregates, comes to the rescue as gap, i.e. `has_gap` is `TRUE`, will increment `group` by one (@sec-numeric-summaries-of-logicals):
```

### Positions

There's one final type of summary that's useful for numeric vectors, but also works with every other type of value: extracting a value at a specific position: `first(x)`, `last(x)`, and `nth(x, n)`.

For example, we can find the first and last departure for each day:

```{r}
flights |> 
  group_by(year, month, day) |> 
  summarize(
    first_dep = first(dep_time, na_rm = TRUE), 
    fifth_dep = nth(dep_time, 5, na_rm = TRUE),
    last_dep = last(dep_time, na_rm = TRUE)
  )
```

> (NB: Because dplyr functions use `_` to separate components of function and arguments names, these functions use `na_rm` instead of `na.rm`.)

Reviewer: ```suggestion
Note that because dplyr functions use `_` to separate components of function and arguments names, these functions use `na_rm` instead of `na.rm`.
```
